### DISCUSSION TITLE
Formula for ConservedLangevin Noise?

### DISCUSSION THREAD

**Discussion by shwetankbasquevolt** [idaholab/moose#26689] [Q&A General]
I am not able to find the formula being used for ConservedLangevin Noise in the docs: [ConservedLangevinNoise](https://mooseframework.inl.gov/source/kernels/ConservedLangevinNoise.html). 
There is a formula given for the non-conserved Langevin noise in the docs: [LangevinNoise](https://mooseframework.inl.gov/source/kernels/LangevinNoise.html) 
I don't see anything similar for Conservedlangevin noise. 
Would appreciate some sources or references for the formula being used for Conserved Langevin noise. I don't have MOOSE installed and I am just trying to replicate a paper in a different software. So, I don't really know how to go through the code to find the formulation.

**Comment by GiudGiud** [comment]
Hello

Please see the source for the kernel here
https://github.com/idaholab/moose/blob/next/modules/phase_field/src/kernels/ConservedLangevinNoise.C

As you can see in the docs for ConservedLangevinNoise, the noise comes from user objects. You can find the source for them here
https://github.com/idaholab/moose/tree/next/modules/phase_field/src/userobjects

Guillaume

**Reply by shwetankbasquevolt** [reply]
Thank you!
